e2e/tests/overview-page.spec.ts

```diff
@@ -238,6 +238,28 @@ test('Typing slash focuses on searchbox', async ({page}) => {
   await expect(searchbox).toHaveAttribute('value', 'def/ghi');
 });
 
+test('Search 50 id search atoms in a query', async ({page}) => {
+  await gotoOverviewPageUrl(page, 'http://localhost:5555/');
+  const searchbox = page.locator('#inputfield');
+  await expect(searchbox).toBeVisible();
+  await expect(searchbox).toHaveAttribute('value', '');
+
+  let query = '';
+  for (let i = 1; i <= 50; i++) {
+    if (!query) {
+      query += 'feature${i}';
+    } else {
+      query += 'OR feature${i}';
+    }
+  }
+  await page.keyboard.type(query);
+  await searchbox.press('Enter');
+
+  // Get all rows.
```

Comment:
Given you are narrowing down to X features, you may want to:
1. count the rows initially to make sure there are more than X rows, then submit the query, then count the rows to make sure there are only X rows.
2. read the number from stats-summary
https://github.com/GoogleChrome/webstatus.dev/blob/18fedc954003b6c67d83a3be42d33baba7322425/frontend/src/static/js/components/webstatus-overview-content.ts#L101-L103 first to make sure there are more than X features, submit the query, and then read stats-summary again.